OMCompiler/SimulationRuntime/c/simulation/solver/newton_diagnostics.c

```diff
@@ -0,0 +1,1280 @@
+/*
+ * This file is part of OpenModelica.
+ *
+ * Copyright (c) 1998-2021, Open Source Modelica Consortium (OSMC),
+ * c/o Linköpings universitet, Department of Computer and Information Science,
+ * SE-58183 Linköping, Sweden.
+ *
+ * All rights reserved.
+ *
+ * THIS PROGRAM IS PROVIDED UNDER THE TERMS OF THE BSD NEW LICENSE OR THE
+ * GPL VERSION 3 LICENSE OR THE OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+ * ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+ * RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+ * ACCORDING TO RECIPIENTS CHOICE.
+ *
+ * The OpenModelica software and the OSMC (Open Source Modelica Consortium)
+ * Public License (OSMC-PL) are obtained from OSMC, either from the above
+ * address, from the URLs: http://www.openmodelica.org or
+ * http://www.ida.liu.se/projects/OpenModelica, and in the OpenModelica
+ * distribution. GNU version 3 is obtained from:
+ * http://www.gnu.org/copyleft/gpl.html. The New BSD License is obtained from:
+ * http://www.opensource.org/licenses/BSD-3-Clause.
+ *
+ * This program is distributed WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE, EXCEPT AS
+ * EXPRESSLY SET FORTH IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE
+ * CONDITIONS OF OSMC-PL.
+ *
+ */
+
+/*! \file newton_diagnostics.c
+ *  Containing all functions to run newton diagnostics on non-linear loops.
+ *  Improve start values for non-linear loops.
+ */
+
+
+/**
+ * @brief Start point for newton diagnostics.
+ *
+ * Calculation of:
+ * 1) alpha     : alpha coefficients
+ * 2) Gamma_ijk : curvature factors
+ * 3) sigma_ij  : solution sensitivities
+ *
+ * According to
+ *    F.Casella and B.Bachman
+ *    On the choice of initial guesses for the Newton-Raphson algorithm
+ *    Applied Mathematics and Computation 398 (2021) 125991
+ *
+ * By Teus van der Stelt, Asimptote bv, the Netherlands
+ * Carried out on behalf of the Delft University of Technology, 2023
+ *
+ * @param data          Pointer to all simulation data.
+ * @param threadData    Pointer to thread data for error handling mainly.
+ */
```

Comment:
Are you documenting the file or a function?
Looks strange. I would suggest to move this to the file Doxygen comment at the top and to remove the inputs.